realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -671,6 +674,517 @@ private void emitCopyOrUpdateMethod(JavaWriter writer) throws IOException {
+                                fieldName)
+                        .endControlFlow()
+                        .emitStatement("Table.nativeSetLink(tableNativePtr, columnInfo.%sIndex, rowIndex, cache%s)", fieldName, fieldName)
+                        .endControlFlow();
+            } else if (Utils.isRealmList(field)) {
+                writer
+                        .emitEmptyLine()
+                        .emitStatement("RealmList<%s> %sList = ((%s) object).%s()",
+                                Utils.getGenericType(field), fieldName, interfaceName, getter)
+                        .beginControlFlow("if (%sList != null)", fieldName)
+                        .emitStatement("long nativeLinkViewPtr = Table.nativeGetLinkView(tableNativePtr, columnInfo.%sIndex, rowIndex)", fieldName)
+                        .beginControlFlow("for (%s %sItem : %sList)", Utils.getGenericType(field), fieldName, fieldName)
+                        .emitStatement("Long cacheItemIndex%s = cache.get(%sItem)", fieldName, fieldName)
+                        .beginControlFlow("if (cacheItemIndex%s == null)", fieldName)
+                        .emitStatement("cacheItemIndex%s = %s.insert(realm, %sItem, cache)", fieldName, Utils.getProxyClassSimpleName(field), fieldName)
+                        .endControlFlow()
+                        .emitStatement("LinkView.nativeAdd(nativeLinkViewPtr, cacheItemIndex%s)", fieldName)
+                        .endControlFlow()
+                        .emitStatement("LinkView.nativeClose(nativeLinkViewPtr)")
+                        .endControlFlow()
+                        .emitEmptyLine();
+
+            } else {
+                setTableValues(writer, fieldType, fieldName, interfaceName, getter, false);
+            }
+        }
+
+        writer.emitStatement("return rowIndex");
+        writer.endMethod();
+        writer.emitEmptyLine();
+    }
+
+    private void emitInsertListMethod(JavaWriter writer) throws IOException {
+        writer.beginMethod(
+                "void", // Return type
+                "insert", // Method name
+                EnumSet.of(Modifier.PUBLIC, Modifier.STATIC), // Modifiers
+                "Realm", "realm", "Iterator<? extends RealmModel>", "objects", "Map<RealmModel,Long>", "cache" // Argument type & argument name
+        );
+
+        writer.emitStatement("Table table = realm.getTable(%s.class)", className);
+        writer.emitStatement("long tableNativePtr = table.getNativeTablePointer()");
+        writer.emitStatement("%s columnInfo = (%s) realm.schema.getColumnInfo(%s.class)"
+                , columnInfoClassName(), columnInfoClassName(), className);
+        writer.emitStatement("%s object = null", className);
+
+        writer.beginControlFlow("while(objects.hasNext())");
+        writer.emitStatement("object = (%s) objects.next()", className);
+
+        writer.emitStatement("long rowIndex = Table.nativeAddEmptyRow(tableNativePtr, 1)");
```

Comment:
Yes, but there are 2 problems with that:

1- You don't know before you traverse the graph how many new rows you need, ex on some cyclic dependency you're going to skip creating a new rows, also in `insertOrUpdate` you may end up updating the row rather than inserting a new one.

2- You need to keep track  of the next available row index  as you may skip some (point 1), at the end AFAIK there's no Core method to deallocate the unused rows, (you may end up iterating through the unused indexes & call delete?)